Sources/Logging/Logging.swift

```diff
@@ -100,17 +100,27 @@ public struct Logger {
 // implementation.
 public enum Logging {
     private static let lock = ReadWriteLock()
-    private static var _factory: (String) -> LogHandler = StdoutLogger.init
+    private static var factory: (String) -> LogHandler = StdoutLogger.init
+    private static var initialized = false
 
     // Configures which `LogHandler` to use in the application.
     public static func bootstrap(_ factory: @escaping (String) -> LogHandler) {
         self.lock.withWriterLock {
-            self._factory = factory
+            precondition(!self.initialized, "logging system can only be initialized once per process")
```

Comment:
Should it also print "currently used factory: \(factory)"? For debugging what the current one is?